src/kvstore/listener/elasticsearch/ESListener.cpp

```diff
@@ -70,7 +54,13 @@ bool ESListener::apply(const BatchHolder& batch) {
     pickTagAndEdgeData(std::get<0>(log), std::get<1>(log), std::get<2>(log), callback);
   }
   if (!bulk.empty()) {
-    auto status = esAdapter_.bulk(bulk);
+    auto esAdapterRes = getESAdapter();
```

Comment:
Why we need to retrieve adapter for each data?

Reply:
Because the password of elasticsearch may be modified during the system operation. Therefore, you need to obtain the configuration of the elasticsearch service again.

In addition, there was a problem before that. If there was no sign in text service when adding listener, the nebula-listener would crash directly. Therefore, the address of elasticsearch cannot be obtained during initialization.

Reply:
👌